cmds/core/ip/ip_linux.go

```
@@ -223,11 +223,12 @@ func parseFlags(args []string, out io.Writer) (cmd, error) {
 			return cmd, fmt.Errorf("failed to parse rcvbuf flag: %w", err)
 		}
 
-		handle.SetSocketReceiveBufferSize(bufSize, true)
+		if err := handle.SetSocketReceiveBufferSize(bufSize, true); err != nil {
+			return cmd, fmt.Errorf("parseFlags: %w", err)
+		}
```

This causes the unit test for ip to fail:

```
 ip_linux_test.go:136: unexpected error: parseFlags: operation not permitted
--- FAIL: TestParseFlags (0.00s)
    --- PASS: TestParseFlags/no_args (0.00s)
    --- PASS: TestParseFlags/inet4 (0.00s)
    --- PASS: TestParseFlags/inet6 (0.00s)
    --- PASS: TestParseFlags/mpls (0.00s)
    --- PASS: TestParseFlags/bridge (0.00s)
    --- PASS: TestParseFlags/link (0.00s)
    --- PASS: TestParseFlags/family (0.00s)
    --- PASS: TestParseFlags/family_inet6 (0.00s)
    --- PASS: TestParseFlags/family_err (0.00s)
    --- PASS: TestParseFlags/resolve (0.00s)
    --- PASS: TestParseFlags/color (0.00s)
    --- FAIL: TestParseFlags/rcvBuf (0.00s
```

In this case, the SetSocketReceiveBufferSize function error is just not checked. It seems that running this with non-root permissions fails, but since the error is not checked, we don't catch this error.